crates/masking/src/masked_string.rs

```diff
@@ -0,0 +1,119 @@
+//!
+//! Secret truncated strings
+//!
+//! Using type aliases for truncating and masking different types of string values.
+
+use alloc::string::{String, ToString};
+
+use super::SerializableSecret;
+
+/// Function for masking alphanumeric characters in a string
+/// examples
+///  Sort Code
+///   (12-34-56, 2, 2) -> 12-**-56
+///  Routing number
+///   (026009593, 3, 3) -> 026***593
+///  CNPJ
+///   (12345678901, 4, 4) -> 1234***8901
+///  Pix key
+///   (123e-a452-1243-1244-000, 4, 4) -> 123e-****-****-****-000
+///  IBAN
+///   (AL35202111090000000001234567, 5, 5) -> AL352******************34567
+fn apply_mask(val: &str, unmasked_char_count: usize, min_masked_char_count: usize) -> String {
+    let len = val.len();
+    if len <= unmasked_char_count {
+        return val.to_string();
+    }
+
+    let mask_start_index =
+    // For showing only last `unmasked_char_count` characters
+    if len < (unmasked_char_count * 2 + min_masked_char_count) {
```

Comment:
Doesn't this condition make `(12345678901, 4, 4) -> 1234***8901` this test case to yield `mask_start_index` as 0 and hence o/p would contain string masked from beginning itself?